tasks/vhost.yml

```diff
@@ -2,9 +2,43 @@
 
 - name: Add apache vhosts configurations
   template:
-    src=vhost.conf.j2
-    dest=/etc/apache2/sites-available/{{ item.servername }}.conf
+    src="{{item.template | default('vhost.conf.j2')}}"
+    dest="/etc/apache2/sites-available/{{ item.server_file_name }}.conf"
     owner=root group=root mode=644
   notify: Restart apache
   with_items: apache_vhosts
   when: apache_vhosts != false
+
+- file: path="{{apache_ssl_path}}" state=directory
+
+- name: Ensure document root exists
+  file: path="{{item.document_root}}" state=directory
+  notify: Restart apache
+  with_items: apache_vhosts
+  when: apache_vhosts != false
+
+- name: Copy key
+  command: cp "{{apache_ssl_key_store}}/{{item.1.key_file}}" "{{apache_ssl_path}}"
```

Comment:
I couldn't get the synchronize module to work inside a loop with delegate_to, so I used remote rsync directly and interpret stdout to give idempotence in the copy.